src/Controllers/GagePageController.ts

```diff
@@ -673,17 +692,150 @@ module StreamStats.Controllers {
                         dailyValues = 0
                     };
                     this.dailyFlow = dailyValues
-                    this.formatData();
+                    this.getRatingCurve();
                 }); 
             }
 
+        public getRatingCurve() {
+            const url = 'https://waterdata.usgs.gov/nwisweb/get_ratings?site_no=' + this.gage.code + '&file_type=exsa'
+            // console.log('getDischargeInfo', url)
+            const request: WiM.Services.Helpers.RequestInfo = new WiM.Services.Helpers.RequestInfo(url, true, WiM.Services.Helpers.methodType.GET, 'json');
+            
+            this.dischargeObj = [];
+            console.log('discharge data', this.dischargeObj)
```

Comment:
```suggestion
            // console.log('discharge data', this.dischargeObj)
```

Reply:
In the function getUSGSMeasured it makes a get request to https://waterdata.usgs.gov/nwis/measurements?site_no=021677181&agency_cd=USGS&format=rdb_expanded. If you click that link you can see there is no data returned. If you go to site 021677181 (33.9486, -81.4706) with your network tab open you will be able to see that request. If you click on that request, then click on the response tab you will see the html returned with the `<title>USGS NwisWeb error message</title>`. Here is an image of my network tab: [image](https://user-images.githubusercontent.com/59063181/228291689-cc576e17-47bb-468f-aa9a-4ceb0c1b241b.png).

I would do something like this in the getUSGSMeasured function to check for this:

 ```
public getUSGSMeasured() {
            const url = 'https://waterdata.usgs.gov/nwis/measurements?site_no=' + this.gage.code + '&agency_cd=USGS&format=rdb_expanded'
            const request: WiM.Services.Helpers.RequestInfo = new WiM.Services.Helpers.RequestInfo(url, true, WiM.Services.Helpers.methodType.GET, 'json');
            this.measuredObj = [];
            this.Execute(request).then(
                (response: any) => {
                    const data = response.data
                    var errorMessage = '<title>USGS NwisWeb error message</title>'
                    var error = data.includes(errorMessage)
                    if (error == false) { // No error
                        const data = response.data.split('\n').filter(r => { return (!r.startsWith("#") && r != "") });
                        data.shift().split('\t');
                        data.shift();
                        data.forEach(row => {
                            let dataRow = row.split('\t')
                            const object = { 
                                dateTime: dataRow[3],
                                timeZone: dataRow[4],
                                quality: dataRow[10],
                                control: dataRow[13],
                                x: parseFloat(dataRow[9]),
                                y: parseFloat(dataRow[8]),
                                qualityColor: this.stageDischargeQualityColor(dataRow[10]),
                                color: this.stageDischargeAgeColor (new Date(dataRow[3])),
                                ageColor: this.stageDischargeAgeColor (new Date(dataRow[3]))
                                // time: this.dateTime (dataRow[3]),
                                // color: this.getQualityCorrectColor (dataRow[10])
                            };
                            this.measuredObj.push(object) 
                        });
                    }
                }, (error) => {
                    // console.log(error)
                }).finally(() => {
                    this.formatData()
                });
        } 
```

 I tested this and it seems to be working. If you add this you should also be able to remove that extra code you added for the legend from this comment https://github.com/USGS-WiM/StreamStats/pull/1576#pullrequestreview-1342359184.